reviewrot/base.py

```diff
@@ -0,0 +1,68 @@
+import datetime
+from dateutil.relativedelta import relativedelta
+
+class BaseService(object):
+    
+    def __init__(self):
+        pass
+    
+    def find_duration(self, rel_diff):
```

Comment:
Also, I was able to simplify this for you:

```python
def format_duration(rel_diff):
    time_dict = OrderedDict([
        ('year', rel_diff.years),
        ('month', rel_diff.months),
        ('day', rel_diff.days),
        ('hour', rel_diff.hours),
        ('minute', rel_diff.minutes),
    ])

    result = []
    for k, v in time_dict.items():
        if v == 1:
            result.append('%s %s' % (v, k))
        elif v >= 1:
            result.append('%s %ss' % (v, k))

    return ' '.join(result)
```

As long as you don't need to support < python 2.7 this will work well.